### DISCUSSION TITLE
FUSE meeting nots: May 21 2024

### DISCUSSION THREAD

**Discussion by orso82** [ProjectTorreyPines/FUSE.jl#568] [General]
**Orso**
* Looking at identifying tasks needed to get functional DIII-D pulse simulator
   1. Populate FUSE's IMAS `pulse_schedule`, `nbi`, `ec`... IDSs with data from DIII-D 
   2. Handle transition from limited to diverted plasmas
   3. Finish wrapping up the RABBIT NBI model
   4. Understand how to handle the NT, L-mode edge and LH-transition
   5. Wrapper around TORBEAM for ECH modeling
   6. Actually exercise time-dependent FUSE on DIII-D
   7.  New free boundary equilibrium solver (FIESTA) in FUSE 
   8. TokSys-FUSE coupling via FXP
   9. Translate TokSys machine/pulse description to IMAS IDSs, and/or viceversa
* Handling of limited plasma now works. Started to work on limited-to-diverted transition

**Galina**
* Found an issue with equilibrium -- got **almost** resolved
* NT studies -- `sat0quench_em_d3d+mastu_azf+1` seems to converge

**Tim**
* Looking at different methods to capture negative triangularity edge:
   1. Use of a "typical" NT inverse scale-length profile
   2. TGLF(-NN) up to 0.98 `sat0quench_em_d3d+mastu_azf+1` ok?
   3. Stored energy fractions

**Adriana**
* Meeting with costing with Nicolai & Paul about LTS Vs HTS costs
   - All SC have different "grade"
   - Nb3Sn & ReBCO are now similar cost when including the coil manufacturing itself
   - HTS still higher risk
   - Quick back of the envelope estimate puts both at $20M/m^3 for full coil
* K-DEMO
   - paper actually says that K-DEMO high Jc superconductor is unproven technology (TRL 2)
* Will work on implementing risk (based on TRL)
   - outlined a risk actor
   - getting TRL for different fusion technologies (ask @bgrierson21 and @daveweisberg)
* Rabbit
   - Total prompt losses seem about ok, but different channels compared to NUBEAM
   - Found the bug that was giving issue with torque... to be fixes soon!

**Nan**
* Impurity, talk to Tomas, and got DIII-D data for carbon
   - 10 cases to try our method on
   - OMFIT for input.profiles --> ods --> input.gacode
   - some Ne data to come after

**Comment by bclyons12** [comment]
A couple quick updates/thoughts from me since I was unable to attend today

- I'm not sure how @orso82 has implemented the limited plasma state. My recommendation would be to track the flux at some number of limiter points (along the first wall, for example) and at the saddle points (X-points) of the flux. Then one point needs to be designated as the "active" boundary point, and a flux surface drawn at that flux value.

- FIESTA (thanks ChatGPT, apologies to stellarator/SIESTA enthusiasts) is coming along. I basically have the innermost solve (inversion of the G-S operator) complete using the Ferrite.jl finite-element package. It can take a dd, mesh the area inside the first wall, and compute the flux from a given `Jt(R,Z)`. Next up is iterations on the nonlinear dependence of `Jt` on the flux. After that, there's another iteration on the boundary value of the flux plus shape/vertical-stability feedback. Below is an example mesh, and then a comparison of the computed and analytic flux for an external coil.

<img width="713" alt="Screenshot 2024-05-21 at 8 50 13 PM" src="https://github.com/ProjectTorreyPines/FUSE.jl/assets/1216060/f6dc18c6-fda0-4ef2-bf05-9c83462afc31">
<img width="709" alt="Screenshot 2024-05-21 at 8 49 42 PM" src="https://github.com/ProjectTorreyPines/FUSE.jl/assets/1216060/cc59a1d1-a822-4df2-849f-42160540111a">
